src/main/java/org/jenkinsci/plugins/workflow/support/steps/stash/UnstashStep.java

```diff
@@ -70,9 +72,9 @@ public String getName() {
             this.name = name;
         }
 
-        @Override protected Void run() throws Exception {
+        @Override protected List<String> run() throws Exception {
             StashManager.unstash(getContext().get(Run.class), name, getContext().get(FilePath.class), getContext().get(TaskListener.class));
-            return null;
+            return StashManager.fileNamesForStash(getContext().get(Run.class), name);
```

Comment:
Do I understand correctly that we're extracting the files twice?  That seems... rather awkward.  Or at least something we might consider making optional and not mandatory with each run.